diff --git a/configure b/configure
index ea413eb..10641f6 100755
--- a/configure
+++ b/configure
@@ -48424,7 +48424,7 @@ then :
   printf %s "(cached) " >&6
 else $as_nop
   r_save_LIBS="${LIBS}"
-LIBS="${LIBS} -licuuc -licui18n"
+LIBS="${LIBS} -licui18n -licuuc -licudata -lstdc++ -lm -ldl"
 if test "$cross_compiling" = yes
 then :
   r_cv_icu=no
@@ -48473,7 +48473,7 @@ if test "x${r_cv_icu}" = xyes; then
 
 printf "%s\n" "#define USE_ICU 1" >>confdefs.h
 
-  LIBS="${LIBS} -licuuc -licui18n"
+  LIBS="${LIBS} -licui18n -licuuc -licudata -lstdc++ -lm -ldl"
 else
   use_ICU=no
 fi
@@ -48575,7 +48575,7 @@ ac_x_includes=no
 ac_x_libraries=no
 # Do we need to do anything special at all?
 ac_save_LIBS=$LIBS
-LIBS="-lX11 $LIBS"
+LIBS="-lX11 -lxcb -lXau -lXdmcp -lSM -lICE -lbsd $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <X11/Xlib.h>
@@ -48717,7 +48717,7 @@ if test "$ac_x_libraries" = no; then
   # See if we find them without any special options.
   # Don't add to $LIBS permanently.
   ac_save_LIBS=$LIBS
-  LIBS="-lX11 $LIBS"
+  LIBS="-lX11 -lxcb -lXau -lXdmcp -lSM -lICE -lbsd $LIBS"
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <X11/Xlib.h>
@@ -48805,6 +48805,7 @@ else
     X_CFLAGS="$X_CFLAGS -I$x_includes"
   fi
 
+  X_LIBS="-lxcb"
   # It would also be nice to do this for all -L options, not just this one.
   if test -n "$x_libraries"; then
     X_LIBS="$X_LIBS -L$x_libraries"
@@ -48888,7 +48889,7 @@ rm -f core conftest.err conftest.$ac_objext conftest.beam \
     # Martyn Johnson says this is needed for Ultrix, if the X
     # libraries were built with DECnet support.  And Karl Berry says
     # the Alpha needs dnet_stub (dnet does not exist).
-    ac_xsave_LIBS="$LIBS"; LIBS="$LIBS $X_LIBS -lX11"
+    ac_xsave_LIBS="$LIBS"; LIBS="$LIBS $X_LIBS -lX11 -lxcb -lXau -lXdmcp -lSM -lICE -lbsd"
     cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -49396,7 +49397,7 @@ then :
   printf %s "(cached) " >&6
 else $as_nop
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lXt ${X_LIBS} -lX11 $LIBS"
+LIBS="-lXt ${X_LIBS} -lX11 -lxcb -lXau -lXdmcp -lSM -lICE -lbsd $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -49448,7 +49449,7 @@ if test "x${use_X11}" = "xyes"; then
 
 printf "%s\n" "#define HAVE_X11 1" >>confdefs.h
 
-  X_LIBS="${X_LIBS} -lX11 -lXt"
+  X_LIBS="${X_LIBS} -lX11 -lxcb -lXt -lXau -lXdmcp -lSM -lICE -lbsd"
 else
   if test "x${with_x}" != "xno"; then
     as_fn_error $? "--with-x=yes (default) and X11 headers/libs are not available" "$LINENO" 5
diff --git a/doc/NEWS.Rd b/doc/NEWS.Rd
index 25bdb12..5a20d31 100644
--- a/doc/NEWS.Rd
+++ b/doc/NEWS.Rd
@@ -7,6 +7,13 @@
 \encoding{UTF-8}
 
 \section{\Rlogo CHANGES IN R 4.3.3}{
+  \subsection{CHANGES IN POSIT'S BUILD FROM \url{https://github.com/r-hub/r-glibc}}{
+    \itemize{
+      \item readRDS() and unserialize() now signal an error instead of
+      returning a PROMSXP, to fix CVE-2024-27322.
+    }
+  }
+
   \subsection{NEW FEATURES}{
     \itemize{
       \item \code{iconv()} now fixes up variant encoding names such as
diff --git a/src/main/serialize.c b/src/main/serialize.c
index 54e4331..4c35749 100644
--- a/src/main/serialize.c
+++ b/src/main/serialize.c
@@ -2583,6 +2583,13 @@ do_serializeToConn(SEXP call, SEXP op, SEXP args, SEXP env)
     return R_NilValue;
 }
 
+static SEXP checkNotPromise(SEXP val)
+{
+    if (TYPEOF(val) == PROMSXP)
+	error(_("cannot return a promise (PROMSXP) object"));
+    return val;
+}
+
 /* unserializeFromConn(conn, hook) used from readRDS().
    It became public in R 2.13.0, and that version added support for
    connections internally */
@@ -2632,7 +2639,7 @@ do_unserializeFromConn(SEXP call, SEXP op, SEXP args, SEXP env)
 	con->close(con);
 	UNPROTECT(1);
     }
-    return ans;
+    return checkNotPromise(ans);
 }
 
 /*
@@ -3263,8 +3270,8 @@ attribute_hidden SEXP
 do_serialize(SEXP call, SEXP op, SEXP args, SEXP env)
 {
     checkArity(op, args);
-    if (PRIMVAL(op) == 2) return R_unserialize(CAR(args), CADR(args));
-
+    if (PRIMVAL(op) == 2) //return R_unserialize(CAR(args), CADR(args));
+	return checkNotPromise(R_unserialize(CAR(args), CADR(args)));
     SEXP object, icon, type, ver, fun;
     object = CAR(args); args = CDR(args);
     icon = CAR(args); args = CDR(args);
diff --git a/src/modules/X11/Makefile.in b/src/modules/X11/Makefile.in
index 0a118d8..ad9fc2a 100644
--- a/src/modules/X11/Makefile.in
+++ b/src/modules/X11/Makefile.in
@@ -20,7 +20,9 @@ ALL_CFLAGS = $(ALL_CFLAGS_LO)
 ## This relies on ld re-ordering the flags, as X_PRE_LIBS does not
 ## contain a -L flag but relies on that in X_LIBS.
 ## This comes from AC_PATH_XTRA so we have not interfered.
-ALL_X_LIBS = $(X_PRE_LIBS) @CAIROX11_LIBS@ $(X_LIBS) $(X_EXTRA_LIBS)
+CAIRO_X_LIBS = @CAIROX11_LIBS@
+CAIRO_X_LIBS2 := $(CAIRO_X_LIBS:-lX11=-Wl,--whole-archive -lX11 -Wl,--no-whole-archive)
+ALL_X_LIBS = $(X_PRE_LIBS) $(CAIRO_X_LIBS2) $(X_LIBS) $(X_EXTRA_LIBS)
 
 SOURCES = devX11.c rotated.c rbitmap.c 
 DEPENDS = $(SOURCES:.c=.d)
