--- a/configure	2025-10-31 09:02:08
+++ b/configure	2025-11-02 23:46:37
@@ -53096,7 +53096,7 @@
   printf %s "(cached) " >&6
 else case e in #(
   e) r_save_LIBS="${LIBS}"
-LIBS="${LIBS} -licuuc -licui18n"
+LIBS="${LIBS} -licui18n -licuuc -licudata -lstdc++ -lm -ldl"
 if test "$cross_compiling" = yes
 then :
   r_cv_icu=no
@@ -53148,7 +53148,7 @@
 
 printf "%s\n" "#define USE_ICU 1" >>confdefs.h
 
-  LIBS="${LIBS} -licuuc -licui18n"
+  LIBS="${LIBS} -licui18n -licuuc -licudata -lstdc++ -lm -ldl"
 else
   use_ICU=no
 fi
@@ -53259,7 +53259,7 @@
 ac_x_libraries=no
 # Do we need to do anything special at all?
 ac_save_LIBS=$LIBS
-LIBS="-lX11 $LIBS"
+LIBS="-lX11 -lXcursor -lXrender -lxcb -lXau -lXdmcp -lSM -lICE -lbsd $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <X11/Xlib.h>
@@ -53402,7 +53402,7 @@
   # See if we find them without any special options.
   # Don't add to $LIBS permanently.
   ac_save_LIBS=$LIBS
-  LIBS="-lX11 $LIBS"
+  LIBS="-lX11 -lXcursor -lXrender -lxcb -lXau -lXdmcp -lSM -lICE -lbsd $LIBS"
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <X11/Xlib.h>
@@ -53492,6 +53492,7 @@
     X_CFLAGS="$X_CFLAGS -I$x_includes"
   fi
 
+  X_LIBS="-lxcb"
   # It would also be nice to do this for all -L options, not just this one.
   if test -n "$x_libraries"; then
     X_LIBS="$X_LIBS -L$x_libraries"
@@ -53577,7 +53578,7 @@
     # Martyn Johnson says this is needed for Ultrix, if the X
     # libraries were built with DECnet support.  And Karl Berry says
     # the Alpha needs dnet_stub (dnet does not exist).
-    ac_xsave_LIBS="$LIBS"; LIBS="$LIBS $X_LIBS -lX11"
+    ac_xsave_LIBS="$LIBS"; LIBS="$LIBS $X_LIBS -lX11 -lXcursor -lXrender -lxcb -lXau -lXdmcp -lSM -lICE -lbsd"
     cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -54156,7 +54157,7 @@
   printf %s "(cached) " >&6
 else case e in #(
   e) ac_check_lib_save_LIBS=$LIBS
-LIBS="-lXt ${X_LIBS} -lX11 $LIBS"
+LIBS="-lXt ${X_LIBS} -lX11 -lXcursor -lXrender -lxcb -lXau -lXdmcp -lSM -lICE -lbsd $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
--- a/src/modules/X11/Makefile.in
+++ b/src/modules/X11/Makefile.in
@@ -20,7 +20,10 @@ ALL_CFLAGS = $(ALL_CFLAGS_LO)
 ## This relies on ld re-ordering the flags, as X_PRE_LIBS does not
 ## contain a -L flag but relies on that in X_LIBS.
 ## This comes from AC_PATH_XTRA so we have not interfered.
-ALL_X_LIBS = $(X_PRE_LIBS) @CAIROX11_LIBS@ $(X_LIBS) $(X_EXTRA_LIBS)
+CAIRO_X_LIBS = @CAIROX11_LIBS@
+# CAIRO_X_LIBS2 := $(patsubst -lX11,-Wl\\,--whole-archive -lX11 -Wl\\,--no-whole-archive,$(CAIRO_X_LIBS))
+CAIRO_X_LIBS2 := $(CAIRO_X_LIBS:-lX11=-Wl,--whole-archive -lX11 -lXcursor -lSM -lICE -Wl,--no-whole-archive -lbsd -luuid -lXrender -lXfixes)
+ALL_X_LIBS = $(X_PRE_LIBS) $(CAIRO_X_LIBS2) $(X_LIBS) $(X_EXTRA_LIBS)

 SOURCES = devX11.c rotated.c rbitmap.c
 DEPENDS = $(SOURCES:.c=.d)
