diff --git a/configure b/configure
index ea413eb..334f852 100755
--- a/configure
+++ b/configure
@@ -48424,7 +48424,7 @@ then :
   printf %s "(cached) " >&6
 else $as_nop
   r_save_LIBS="${LIBS}"
-LIBS="${LIBS} -licuuc -licui18n"
+LIBS="${LIBS} -licui18n -licuuc -licudata -lstdc++ -lm -ldl"
 if test "$cross_compiling" = yes
 then :
   r_cv_icu=no
@@ -48473,7 +48473,7 @@ if test "x${r_cv_icu}" = xyes; then
 
 printf "%s\n" "#define USE_ICU 1" >>confdefs.h
 
-  LIBS="${LIBS} -licuuc -licui18n"
+  LIBS="${LIBS} -licui18n -licuuc -licudata -lstdc++ -lm -ldl"
 else
   use_ICU=no
 fi
